File: android/src/main/kotlin/pers/hal42/util/Ticks.java
package pers.hal42.util;

/**
 * Java timing is mostly done in milliseconds.
 * This class wraps converting human units into milliseconds.
 * It may seem gratuitous, but removes embedded constants.
 */

public class Ticks {

  public static final long perSecond = 1000L;
  public static final long perMinute = 60 * perSecond;
  public static final long perHour = 60 * perMinute;
  public static final long perDay = 24 * perHour;

  public static final long forSeconds(int seconds) {
    return seconds * perSecond;
  }

  public static final long forSeconds(double seconds) {
    return (long) (seconds * ((double) perSecond));
  }

  public static final long forMinutes(int minutes) {
    return minutes * perMinute;
  }

  public static final long forHours(int hours) {
    return hours * perHour;
  }

  public static final long forDays(int days) {
    return days * perDay;
  }

}

File: android/src/main/kotlin/pers/hal42/util/Ticks.kt
package pers.hal42.util

/**
 * Java timing is mostly done in milliseconds.
 * This class wraps converting human units into milliseconds.
 * It may seem gratuitous, but removes embedded constants.
 */

object Ticks {

  const val perSecond = 1000L
  const val perSecondd = 1000F
  const val perMinute = 60 * perSecond
  const val perHour = 60 * perMinute
  const val perDay = 24 * perHour

  fun forSeconds(seconds: Int): Long {
    return seconds * perSecond
  }

  fun forSeconds(seconds: Double): Long {
    return (seconds * perSecondd).toLong()
  }

  fun forMinutes(minutes: Int): Long {
    return minutes * perMinute
  }

  fun forHours(hours: Int): Long {
    return hours * perHour
  }

  fun forDays(days: Int): Long {
    return days * perDay
  }

}
